### PR TITLE
Add thrift support for Operator Info

### PR DESCRIPTION
**Pull request by ajaygeorge**
- Add OperatorInfoUnion
- Add thrift support for OperatorInfo implementations
- Add thrift support for OperatorStats

Test plan - Unit test and manual testing

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java [hunk 1]
```diff
@@ -85,7 +89,9 @@
     private final Optional<BlockedReason> blockedReason;
 
     @Nullable
-    private final OperatorInfo info;
+    private OperatorInfo info;
+    @Nullable
+    private OperatorInfoUnion infoUnion;
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java:94]: Just assign one or the other to null so we can keep these as immutable?

```suggestion
    private final OperatorInfo info;
    @Nullable
    private final OperatorInfoUnion infoUnion;
```

**Review by tdcmeehan — COMMENTED** [review]
Should we mark `OperatorInfo` as deprecated, with a comment indicating we're moving to Thrift?

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java (hunk 1 above)
Comment (ajaygeorge) [reply presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java:94]: as discussed offline, OperatorInfo is used in multiple places and is an interface. The plan is not to deprecate the class immediately.

Comment (ajaygeorge) [reply presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java]: done. Added final modifiers.

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorInfoUnion.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            return new OperatorInfoUnion((WindowInfo) info);
+        }
+        else if (info instanceof TableWriterOperator.TableWriterInfo) {
+            return new OperatorInfoUnion((TableWriterOperator.TableWriterInfo) info);
+        }
+        else if (info instanceof TableWriterMergeInfo) {
+            return new OperatorInfoUnion((TableWriterMergeInfo) info);
+        }
+        else {
+            throw new IllegalArgumentException("OperatorInfo is of an unknown type");
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorInfoUnion.java:218]: Perhaps also print the classname of the unknown type?  Ditto below

Comment (ajaygeorge) [reply presto-main/src/main/java/com/facebook/presto/operator/OperatorInfoUnion.java]: done

**Review by tdcmeehan — APPROVED** [review]

**Comment by tdcmeehan** [comment]
Any other comments?

**Pull request merged**